contracts/anchor-handler/src/contract.rs

```diff
@@ -0,0 +1,345 @@
+        } => exec_set_resource(deps, info, resource_id, contract_addr),
+        ExecuteMsg::MigrateBridge { new_bridge } => migrate_bridge(deps, info, new_bridge),
+
+        /* ---  Anchor-handler specific execution entries --- */
+        // Proposal execution should be initiated when a proposal is finalized in the Bridge contract.
+        // by a relayer on the deposit's destination chain
+        ExecuteMsg::ExecuteProposal { resource_id, data } => {
+            execute_proposal(deps, info, resource_id, data)
+        }
+    }
+}
+
+fn exec_set_resource(
+    deps: DepsMut,
+    info: MessageInfo,
+    resource_id: [u8; 32],
+    contract_addr: String,
+) -> Result<Response, ContractError> {
+    let state = STATE.load(deps.storage)?;
+
+    // Validations
+    if info.sender != state.bridge_addr {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Save/update the mapping `resource_id => contract_addr`
+    let contract_addr = deps.api.addr_validate(&contract_addr)?;
+    set_resource(deps.storage, resource_id, contract_addr)?;
+
+    Ok(Response::new().add_attribute("method", "set_resource"))
+}
+
+fn migrate_bridge(
+    deps: DepsMut,
+    info: MessageInfo,
+    new_bridge: String,
+) -> Result<Response, ContractError> {
+    let state = STATE.load(deps.storage)?;
+
+    // Validations
+    if info.sender != state.bridge_addr {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Migrage(update) the "bridge_addr" with "new_bridge"
+    let bridge_addr = deps.api.addr_validate(&new_bridge)?;
+    STATE.save(deps.storage, &State { bridge_addr })?;
+
+    Ok(Response::new().add_attribute("method", "migrate_bridge"))
+}
+
+fn execute_proposal(
+    deps: DepsMut,
+    info: MessageInfo,
+    resource_id: [u8; 32],
+    data: Vec<u8>,
+) -> Result<Response, ContractError> {
+    // Parse the (proposal)`data`.
+    let parsed_resource_id = element_encoder(&data[0..32]);
+    let func_sig = bytes4_encoder(&data[32..36]);
+    let arguments = &data[36..];
+
+    let state = STATE.load(deps.storage)?;
+
+    // Validations
+    if info.sender != state.bridge_addr {
+        return Err(ContractError::Unauthorized {});
+    }
+    if parsed_resource_id != resource_id {
+        return Err(ContractError::Std(StdError::GenericErr {
+            msg: "Invalid resource id".to_string(),
+        }));
+    }
+    let anchor_addr = read_contract_addr(deps.storage, resource_id)?;
+    if !read_whitelist(deps.storage, anchor_addr.clone())? {
+        return Err(ContractError::Std(StdError::GenericErr {
+            msg: "provided tokenAddress is not whitelisted".to_string(),
+        }));
+    }
+
+    // Execute the proposal according to function signature
+    let set_handler_sig = bytes4_encoder(
+        &Keccak256::hash(b"SetHandler(String,u32)").map_err(|_| ContractError::DecodeError)?,
+    );
+    let update_edge_sig = bytes4_encoder(
+        &Keccak256::hash(b"UpdateEdge(u64,[u8;32],u32,[u8;32])")
```

Comment:
These should match the function itself. It should be the exact snake_case value not the camelCase proposal.